mlir/lib/Dialect/Rock/Transforms/ConvToGemm.cpp

```diff
@@ -171,6 +172,11 @@ static int64_t getUtilityVectorizationLen(ShapedType shape,
                         kMaxVectorOpLen * elemsPerWord);
 }
 
+static bool isLargeTensor(ShapedType type) {
+  return (type.getNumElements() * (type.getElementTypeBitWidth() / 8)) >=
+         ((int64_t)1 << (int64_t)32);
```

Comment:
I don't think there's a numeric_limits for 0x1_00_00_00_00

Reply:
ah ha... my bad then... why that value though ? just curious

Reply:
hold on isn;t this equivalent to :  > std::numeric_limits<int32_t>::max()
Note : >= vs >

Reply:
hence the duplication from below

Reply:
... no, it would be `> std::numeric_limits<uint32_t>::max()` ... note the u, and even then we'd need to cast to a longer type, so I might as well ignore that and write out 4 GB as a constant.

Reply:
I'm not seeing the problem with `2 ^ 32` directly?

Reply:
... but yeah, is a minor point, I can fix